examples/go/Dockerfile

```diff
@@ -1,12 +1,28 @@
-FROM golang:1.12 as build-env
+# our builder image
+FROM golang:1.13.1-alpine AS builder
 
+# default app directory
 WORKDIR /go/src/app
+
+# copy-in our module info
+COPY go.mod go.sum /go/src/app/
+RUN go mod download
+
+# add our binary sources
 ADD . /go/src/app
 
-RUN go get -d -v ./...
+# build our binary
+RUN CGO_ENABLED=0 go build -a -installsuffix cgo -ldflags '-w -s -extldflags "-static"' -o /go/bin/app
+
+# ---
+
+FROM gcr.io/distroless/static
+
+# copy our built binary
+COPY --from=builder --chown=nonroot /go/bin/app /app
 
-RUN go build -o /go/bin/app
+# run as unprivileged user
+USER nonroot
```

Comment:
Use `gcr.io/distroless/static:nonroot` instead?

Reply:
@chanseokoh I am not a big fan of using tags for build variants, I much prefer explicit security. As long as distroless does not have a clear update/break policy, I keep the tags plain.